File: shared_flutter/lib/src/api/models/register_retailer_request.dart
class RegisterRetailerRequest {
  RegisterRetailerRequest({
    required this.ownerPhone,
    required this.pin,
    this.ownerName,
    required this.shopName,
    required this.shopPhone,
    required this.shopLatitude,
    required this.shopLongitude,
    this.shopAddress,
  });

  final String ownerPhone;
  final String pin;
  final String? ownerName;
  final String shopName;
  final String shopPhone;
  final double shopLatitude;
  final double shopLongitude;
  final String? shopAddress;

  Map<String, dynamic> toJson() => {
        'ownerPhone': ownerPhone,
        'pin': pin,
        'ownerName': ownerName,
        'shopName': shopName,
        'shopPhone': shopPhone,
        'shopLatitude': shopLatitude,
        'shopLongitude': shopLongitude,
        'shopAddress': shopAddress,
      };
}

File: backend/src/GasFinder.Shared/Contracts/Auth/RegisterRetailerRequest.cs
namespace GasFinder.Shared.Contracts.Auth;

public record RegisterRetailerRequest(
    string OwnerPhone,
    string Pin,
    string? OwnerName,
    string ShopName,
    string ShopPhone,
    double ShopLatitude,
    double ShopLongitude,
    string? ShopAddress
);
